index.js

```diff
@@ -39,35 +43,62 @@ function Heartbeat(baseUrl, pulse = 60) {
-  
+
   if (typeof threshold === 'function') callback = threshold;
 
+
+  var beatId = host + '-' + category + '-' + type + '-' + name;
+  var lastPulse = _.has(this._lastPulse, beatId) ? this._lastPulse[beatId] : 0;
+  var thresholdHrs = 25
+  if (category === "manfred") {
+    thresholdHrs = 0
+  }
   // only pulse in production mode
   if (process.env.PRODUCTION === undefined
-  || !(/^(?:yes|true|1|on)$/i).test(process.env.PRODUCTION.toString())) {
+    || !(/^(?:yes|true|1|on)$/i).test(process.env.PRODUCTION.toString())) {
     console.log('<aws-heartbeat-client> [NOTICE] supressing heartbeat pulse outside of production mode');
     if (typeof callback === 'function') callback(null, { result: 'aborted' });
     return false;
   }
-
-  var beatId = host + '-' + category + '-' + type + '-' + name;
-  var lastPulse = _.has(this._lastPulse, beatId) ? this._lastPulse[beatId] : 0;
-
   if (Date.now() - lastPulse < this._pulse * 1000) {
     if (typeof callback === 'function') callback(null, { result: 'skipped' });
 
   } else {
 
     this._lastPulse[beatId] = Date.now();
-    var slash = /\/$/.test(this._baseUrl) ? '' : '/';
-    var url = this._baseUrl + slash + 'pulse?host=' + host + '&category=' + category + '&type=' + type + '&name=' + name;
-    if (threshold && typeof threshold !== 'function') url += '&threshold=' + threshold;
-    request(url, function (err, res, body) {
-      if (err) (typeof callback === 'function') ? callback(err) : console.error(err);
-      else if (res && res.statusCode >= 400) (typeof callback === 'function') ? callback(new Error(res.statusCode + ' ' + res.statusMessage)) : console.log(res.statusMessage);
-      else (typeof callback === 'function') ? callback(null, body) : console.log(body);
-    });
-
+    const query = gql`
+          mutation createHeartbeat($name: String!, $category: String!, $host: String!, $type: String!,$lastSuccessAt: DateTime!, $thresholdHrs: Int!, $status: AlertStatus!){
+            updateCreateHeartbeat(input: { hostName: $host, category: $category, type: $type, name: $name ,lastSuccessAt: $lastSuccessAt, thresholdHrs: $thresholdHrs, status: $status}) {
+          ... on LegacyHeartbeatType {
+                id
+                hostName
+                category
+                type
+                name
+                lastSuccessAt
+                thresholdHrs
+                status
+              }
+            }
+          }
+          `
+    const variables = {
+      name: name,
+      category: category,
+      host: host,
+      type: type,
+      lastSuccessAt: new Date(Date.now()),
+      thresholdHrs: thresholdHrs,
+      status: "CLOSED"
+    }
+    graphQLClient.request(query, variables)
+      .then((data) => {
+        (typeof callback === 'function') ? callback(null, { statusCode: 200, body: "Successful Heartbeat pulse" }) : console.log("Successful Heartbeat pulse")
+      })
+      .catch((err) => {
+        console.log('error' + `${err}`)
```

Comment:
you will log the error twice

Reply:
yep, good catch. forgot to remove that one